tests/v2/validation/provisioning/airgap/rke1_custom_cluster_test.go

```diff
@@ -0,0 +1,126 @@
+	provisioning "github.com/rancher/rancher/tests/framework/extensions/provisioning"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioninginput"
+	nodepools "github.com/rancher/rancher/tests/framework/extensions/rke1/nodepools"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2/validation/pipeline/rancherha/corralha"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning/permutations"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning/registries"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type AirGapRKE1CustomClusterTestSuite struct {
+	suite.Suite
+	client         *rancher.Client
+	session        *session.Session
+	corralPackage  *corral.CorralPackages
+	clustersConfig *provisioninginput.Config
+	registryFQDN   string
+}
+
+func (a *AirGapRKE1CustomClusterTestSuite) TearDownSuite() {
+	a.session.Cleanup()
+}
+
+func (a *AirGapRKE1CustomClusterTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	a.session = testSession
+
+	a.clustersConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, a.clustersConfig)
+
+	corralRancherHA := new(corralha.CorralRancherHA)
+	config.LoadConfig(corralha.CorralRancherHAConfigConfigurationFileKey, corralRancherHA)
+
+	registriesConfig := new(registries.Registries)
+	config.LoadConfig(registries.RegistriesConfigKey, registriesConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(a.T(), err)
+
+	a.client = client
+	listOfCorrals, err := corral.ListCorral()
+	require.NoError(a.T(), err)
+
+	corralConfig := corral.CorralConfigurations()
+
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(a.T(), err)
+
+	a.corralPackage = corral.CorralPackagesConfig()
+
+	_, corralExist := listOfCorrals[corralRancherHA.Name]
+	if corralExist {
+		bastionIP, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryIP)
+		require.NoError(a.T(), err)
+
+		err = corral.UpdateCorralConfig(corralBastionIP, bastionIP)
+		require.NoError(a.T(), err)
+
+		registryFQDN, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryFQDN)
+		require.NoError(a.T(), err)
+		logrus.Infof("registry fqdn is %s", registryFQDN)
+		a.registryFQDN = registryFQDN
+	} else {
+		a.registryFQDN = registriesConfig.ExistingNoAuthRegistryURL
+	}
+
+}
+
+func (a *AirGapRKE1CustomClusterTestSuite) TestProvisioningRKE1CustomCluster() {
+	a.clustersConfig.NodesAndRolesRKE1 = []nodepools.NodeRoles{provisioninginput.RKE1AllRolesPool}
+
+	tests := []struct {
+		name   string
+		client *rancher.Client
+	}{
+		{provisioninginput.AdminClientName.String() + "-" + permutations.RKE1AirgapCluster + "-", a.client},
```

Comment:
Is this fine in terms of regex? @igomez06 

Also, I am not quite sure whether to have a dash. Don't the spaces appended as underscore?

Reply:
will wait for Israel's feedback

Reply:
It doesn't really matter, but the go test already adds "_" so this is kind of redundant.